### PR TITLE
build: bump ts to v5

### LINKED ISSUE #159: 🚀 Feature: Support TypeScript v5
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

new TypeScript version dropping

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by RebeccaStevens**
## PR Checklist

- [x] Addresses an existing open issue, fixes #159
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/ts-api-utils/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/ts-api-utils/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

To be (update and) merged once v5 full comes out.

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]
Love it! Thanks @RebeccaStevens! 🚀

**Comment by RebeccaStevens** [comment]
Something seems to be going wrong with `ts.createProgram`.

**Comment by RebeccaStevens** [comment]
@JoshuaKGoldberg I got this working by using `tsvfs.createVirtualTypeScriptEnvironment` instead of `tsvfs.createVirtualCompilerHost`. Not sure why exactly this change was needed but it works now.

File: tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
 		"declaration": true,
 		"declarationMap": true,
 		"esModuleInterop": true,
-		"moduleResolution": "node",
+		"module": "ESNext",
+		"moduleResolution": "bundler",
```
Comment (RebeccaStevens) [inline tsconfig.json:7]: Note that the current version of vscode doesn't support `"moduleResolution": "bundler"`. The insiders version does though so the stable version should soon.

**Review by JoshuaKGoldberg — APPROVED** [review]
If it works, it works! 🚀

**Pull request merged**